### PR TITLE
feat: implemented the workspace issues table

### LINKED ISSUE #3322: Feature: Issues Table
As a user of workspaces, I want an activity page to display a grid of all the issues for all tracked repositories in a workspace so that I can have a more granular view of issues.

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Adds an issues table to the activity tab of a workspace.

~Note: You won't be able to test this until the API endpoint for it has been merged (see https://github.com/open-sauced/api/pull/774).~

## Related Tickets & Documents

Close #3322

## Mobile & Desktop Screenshots/Recordings

CleanShot 2024-05-09 at 11 34 19

https://github.com/open-sauced/app/assets/833231/f4aed7c0-613b-4095-8ba5-f0ba83b84019

On mobile the heart reactions count only appears if it's greater than 0

CleanShot 2024-05-09 at 11 47 04

## Steps to QA

1. Go to any workspace
2. Click on the Activity tab
3. Click on the Issues tab
4. The issues table loads

## Tier (staff will fill in)

- [ ] Tier 1
- [x] Tier 2
- [ ] Tier 3
- [ ] Tier 4

### CODE REVIEW

**Comment by nickytonline** [comment]
We have reactions available as well in the data @bdougie @isabensusan. Do you think it's worth surfacing them with their counts for each issue?

**Comment by bdougie** [comment]
> We have reactions available as well in the data @bdougie @isabensusan. Do you think it's worth surfacing them with their counts for each issue?

That would be great. One of the things the GitHub cli/cli does is sort by most hearts. Literally how they prioritize their roadmap.

**Comment by nickytonline** [comment]
> > We have reactions available as well in the data @bdougie @isabensusan. Do you think it's worth surfacing them with their counts for each issue?
> 
> That would be great. One of the things the GitHub cli/cli does is sort by most hearts. Literally how they prioritize their roadmap.
> 
> Also maybe the hearts.dev squad would find that useful
> 
> Cc @babblebey @gr2m

You can sort by hearts now. Just wondering if it makes sense to surface or sort by any other reactions like plus one or minus one, eyeballs etc.

**Comment by nickytonline** [comment]
The sub tabs have been updated @bdougie @isabensusan.

CleanShot 2024-05-09 at 11 33 33

**Comment by nickytonline** [comment]
Just sorting out the E2E tests and it would be good to get #3352 merged first.

File: components/TabList/tab-list.tsx [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
 
-interface NavProps {
+interface TabListProps {
   tabList: TabItem[];
   selectedTab: string;
   pageId?: string;
 }
 
-const TabList: React.FC<NavProps> = ({ tabList, pageId, selectedTab }) => {
+interface SubTabsListProps extends TabListProps {
```
Comment (nickytonline) [inline components/TabList/tab-list.tsx:18]: These changes will disappear from this PR once  #3352 is merged.

File: components/Workspaces/TrackedRepositoryFilter.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ export default function TrackedRepositoryFilter({ options, handleSelect }: Track
 
   return (
     <MultiSelect
-      placeholder="Filter"
+      placeholder="All Repositories"
```
Comment (nickytonline) [inline components/Workspaces/TrackedRepositoryFilter.tsx:19]: Small copy change requestwed from @isabensusan.

File: components/Workspaces/WorkspaceIssuesTable.tsx [hunk 3]
```diff
@@ -0,0 +1,308 @@
+import {
```
Comment (nickytonline) [inline components/Workspaces/WorkspaceIssuesTable.tsx:1]: This is pretty much the same thing as the PRs table feature, aside from the different data rendered.

**Review by bdougie — APPROVED** [review]
Thanks for working on this @nickytonline

**Review by brandonroberts — APPROVED** [review]
Works great! I had to reselect repos after switching between tabs but no big deal.

**Pull request merged**